pkg/test/framework/components/istio/istio.go

```diff
@@ -31,8 +31,48 @@ type Instance interface {
 // SetupConfigFn is a setup function that specifies the overrides of the configuration to deploy Istio.
 type SetupConfigFn func(cfg *Config)
 
+func GetDeployedNamespaces(cfg Config) []string {
+	// Store in map to dedupe
+	allNamespace := map[string]struct{}{}
+
+	if cfg.AlphaInstaller {
+		for _, c := range cfg.InstallComponents {
+			switch c {
+			case Base:
+				allNamespace[cfg.IstioNamespace] = struct{}{}
+				allNamespace[cfg.ConfigNamespace] = struct{}{}
+			case Ingress:
+				allNamespace[cfg.IngressNamespace] = struct{}{}
+			case Egress:
+				allNamespace[cfg.EgressNamespace] = struct{}{}
+			case Telemetry:
+				allNamespace[cfg.TelemetryNamespace] = struct{}{}
+			case Policy:
+				allNamespace[cfg.PolicyNamespace] = struct{}{}
+			case Tracing:
+				allNamespace[cfg.TelemetryNamespace] = struct{}{}
+			case Nodeagent:
+				allNamespace[cfg.IstioNamespace] = struct{}{}
+			}
+		}
+	} else {
+		allNamespace[cfg.IstioNamespace] = struct{}{}
+		allNamespace[cfg.ConfigNamespace] = struct{}{}
+		allNamespace[cfg.IngressNamespace] = struct{}{}
+		allNamespace[cfg.EgressNamespace] = struct{}{}
+		allNamespace[cfg.TelemetryNamespace] = struct{}{}
+		allNamespace[cfg.PolicyNamespace] = struct{}{}
+	}
+
+	result := []string{}
```

Comment:
Note that "var result []string" will work just as well for you, but is not strictly the same thing. Given you don't need to rely on the distinction, starting with nil ("var result []string") may be a more natural thing to do, doesn't pose a question why do you need to distinguish empty slice from nil. Though it seems there's a lot of existing code that uses explicit []string{}, so maybe the subtlety is already lost in this codebase.